Simplified Async Autonomous Actions

jmblixt3 opened the pull request.

Isopod00 reviewed (changes requested):
This PR is mostly working but canceling seems to be weird, maybe it's a weird interaction between canceling and the async_sleep() method or something. Whenever I cancel an autonomous action it still seems to be running in the background or something and it is possible to get multiple running at once if I start and cancel an action repeatedly, because canceling does not seem to be fully canceling.

jmblixt3 commented:
> This PR is mostly working but canceling seems to be weird, maybe it's a weird interaction between canceling and the async_sleep() method or something. Whenever I cancel an autonomous action it still seems to be running in the background or something and it is possible to get multiple running at once if I start and cancel an action repeatedly, because canceling does not seem to be fully canceling.

This appears to be fixed now for all actions, give it another try! I think the difference we had was I hadn't tried to cancel during sleep, only skimmer movement

jmblixt3 committed:
Make action utilize async await syntax
Replace all rclpy.spin_until loops with await.
In a few places this wasn't possible so a custom
future callback was used.
The only remaining spin_once is for motor ramp up/
down which is set to be implemented soon
Also removed minor deprecated devcontainer config
for python language server

Signed-off-by: Jonathan Blixt <jmblixt3@gmail.com>

Isopod00 reviewed (approved):
Everything appears to be working now!
